segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/StorageLoader.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host;
+
+import io.pravega.common.ConfigSetup;
+import io.pravega.segmentstore.storage.StorageFactory;
+import java.util.ServiceLoader;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class StorageLoader {
+    public StorageFactory load(ConfigSetup setup, String storageImplementation, ScheduledExecutorService executor) {
+        ServiceLoader<StorageFactory> loader = ServiceLoader.load(StorageFactory.class);
```

Comment:
While it might seem ridiculous, I think it might simplify things if you loaded a StorageFactoryFactory then instead of initialize, you could call createFactory() passing in setup and an executor.

Reply:
The problem is that the serviceloader does not allow constructor with parameters. So even in case of `FactoryFactory`, an `initialize` method will be required. We might as well keep this in the `Factory`.

Reply:
Fixed. Introduced `FactoryFactory`